lib/diaspora/mentions_container.rb

```diff
@@ -0,0 +1,41 @@
+module Diaspora
+  module MentionsContainer
+    extend ActiveSupport::Concern
+
+    included do
+      after_create :create_mentions
+      has_many :mentions, as: :mentions_container, dependent: :destroy
+    end
+
+    def local_mentions
```

Comment:
I would add a scope `local` to the `Mention` model, so you can use `mentionable.mentions.local`